Migrate the SQLite code from the main laminas-db repo to this one

settermjd opened the pull request:
This PR includes @simon-mundy's core work, and some extra clean up after running the QA tools on the code base.

simon-mundy committed:
Initial commit

simon-mundy committed:
Initial commit

simon-mundy committed:
Refactored to remove redundant code

simon-mundy committed:
Initial commit

simon-mundy committed:
Further coverage on Unit/Integration testing

settermjd committed:
Clean up the code base to comply with the coding standard

settermjd committed:
Add and update the QA configuration files

This change updates PHPUnit's configuration file to match the project,
and adds config files for PHP Code_Sniffer and Psalm.

settermjd committed:
Clean up the code in response to static analysis review

These changes, while by no means complete, are a start along the way to
cleaning up the codebase so that it is as simple as possible to perform
static analysis on.

tyrsson reviewed (dismissed):
There is a considerable amount of code that will be removed once its refactored for factories.

settermjd committed:
Update the code to improve static analysis quality

This change, similar to previous ones, contains a series of changes
designed to improve the quality of static analysis with Psalm (and code
readability and maintainability). I don't believe that there are,
effectively, any actual code changes.

settermjd committed:
Update Psalm's configuration

This change resets Psalm's baseline, and updates its configuration to
ignore a series of rules that are, currently, not required, as they're
making the transition more challenging. Specifically, they can be
ignored for the time being while the code is cleaned up, after which
they'll be reinstated.

settermjd committed:
Add static analysis to the Composer QA job

This adds Psalm to the QA job, so that it's included in that task,
ensuring that the code is as maintainable as possible.

settermjd commented:
Sorry about the sheer number of files in this PR.

settermjd commented:
> There is a considerable amount of code that will be removed once its refactored for factories.

Would love to discuss this fully.

tyrsson commented:
I'm happy too anytime we're both online.

tyrsson commented on src/Driver/Pdo/Connection.php:
```diff
@@ -0,0 +1,332 @@
+                        $pdoDriver = substr($pdoDriver, 3) ?: '';
+                    }
+                    break;
+                case 'pdodriver':
```
I missed this in my initial review. I think it would serve us well to remove the one off parameters in favor of using just the 'driver' key. One of my goals is to standardize this to ease documenting all of the changes. I can not confirm this but my hunch is that we find these one offs due to so many different contributors providing code over the years.

tyrsson committed:
Merge pull request #4 from php-db/0.0.x

Update README.md
